error_test.go

```diff
@@ -326,17 +326,43 @@ func TestAll(t *testing.T) {
 		assert.Equal(t, "e2: e1: e0", appErr.FullMessage())
 		assert.Equal(t, nil, appErr.Code)
 		assert.Equal(t, false, appErr.Ignorable)
-		assert.NotEmpty(t, appErr.StackTrace)
-		assert.Equal(t, "errFunc1", appErr.StackTrace[0].Func)
+		assert.Equal(t, []string{
+			"errFunc1",
+			"errFunc2",
+			"errFunc3",
+			"TestAll",
+			"tRunner",
+		}, funcNamesFromStackTrace(appErr.StackTrace))
 	}
 
 	{
 		appErr := Unwrap(errFunc4())
 		assert.Equal(t, "e4: e2: e1: e0", appErr.FullMessage())
 		assert.Equal(t, 500, appErr.Code)
 		assert.Equal(t, true, appErr.Ignorable)
-		assert.NotEmpty(t, appErr.StackTrace)
-		assert.Equal(t, "errFunc1", appErr.StackTrace[0].Func)
+		assert.Equal(t, []string{
+			"errFunc1",
+			"errFunc2",
+			"errFunc3",
+			"errFunc4",
+			"TestAll",
+			"tRunner",
+		}, funcNamesFromStackTrace(appErr.StackTrace))
+	}
+
+	{
+		appErr := Unwrap(errFunc4Goroutine())
+		assert.Equal(t, "e4: e2: e1: e0", appErr.FullMessage())
+		assert.Equal(t, 500, appErr.Code)
+		assert.Equal(t, true, appErr.Ignorable)
+		assert.Equal(t, []string{
+			"errFunc1",
+			"errFunc2",
+			"errFunc3Goroutine.func1",
```

Comment:
Even when the error is passed between goroutines, a virtual stack trace contains all information.